packages/extension-chakra-storefront/src/pages/product-list/index.jsx

```diff
@@ -5,855 +5,137 @@
-import {isHydrated} from '../../utils/utils'
-
-// NOTE: You can ignore certain refinements on a template level by updating the below
-// list of ignored refinements.
-const REFINEMENT_DISALLOW_LIST = ['c_isNew']
+import {useExtensionConfig} from '../../hooks'
+import {useProductListWishlist} from './hooks/use-product-list-wishlist'
+import {useProductListData} from './hooks/use-product-list-data'
+import {useProductListControls} from './hooks/use-product-list-controls'
 
 /*
  * This is a simple product listing page. It displays a paginated list
  * of product hit objects. Allowing for sorting and filtering based on the
  * allowable filters and sort refinements.
  */
-const ProductList = (props) => {
-    // Using destructuring to omit properties; we must rename `isLoading` because we use a different
-    // `isLoading` later in this function.
-    // eslint-disable-next-line react/prop-types, @typescript-eslint/no-unused-vars
-    const {isLoading: _unusedIsLoading, staticContext, ...rest} = props
-    const {formatMessage} = useIntl()
-    const navigate = useNavigation()
-    const history = useHistory()
-    const params = useParams()
-    const location = useLocation()
-    const toast = useToast()
-    const einstein = useEinstein()
-    const dataCloud = useDataCloud()
-    const activeData = useActiveData()
-    const {res} = useServerContext()
-    const customerId = useCustomerId()
-    const [searchParams, {stringify: stringifySearchParams}] = useSearchParams()
+const ProductList = () => {
     const {
-        pages: {ProductList: productListConfig},
-        maxCacheAge: MAX_CACHE_AGE,
-        staleWhileRevalidate: STALE_WHILE_REVALIDATE
+        pages: {ProductList: productListConfig}
     } = useExtensionConfig()
-    /**************** Page State ****************/
-    const [filtersLoading, setFiltersLoading] = useState(false)
-    const [wishlistLoading, setWishlistLoading] = useState([])
-    const [sortOpen, setSortOpen] = useState(false)
-
-    const urlParams = new URLSearchParams(location.search)
-    let searchQuery = urlParams.get('q')
-    const isSearch = !!searchQuery
-
-    if (params.categoryId) {
-        searchParams._refine.push(`cgid=${params.categoryId}`)
-    }
-
-    /**************** Mutation Actions ****************/
-    const {mutateAsync: createCustomerProductListItem} = useShopperCustomersMutation(
-        'createCustomerProductListItem'
-    )
-    const {mutateAsync: deleteCustomerProductListItem} = useShopperCustomersMutation(
-        'deleteCustomerProductListItem'
-    )
-
-    /**************** Query Actions ****************/
-    // _refine is an invalid param for useProductSearch, we don't want to pass it to API call
-    const {_refine, ...restOfParams} = searchParams
+    const {toggleItem, isItemInWishlist} = useProductListWishlist()
 
     const {
-        isLoading,
+        category,
+        filtersLoading,
+        handleProductClick,
         isFetched,
+        isLoading,
```

Comment:
Not directly related to this PR. Should we replace isLoading with isPending - React query 5 has changed isLoading to isPending. https://github.com/TanStack/query/discussions/6297